docs: revocation_endpoint (#2654)
caos/zitadel: docs/docs/apis/openidoauth/endpoints.md

If the token is invalid or expired, an HTTP 401 will be returned.

## revocation_endpoint

[https://api.zitadel.ch/oauth/v2/revoke](https://api.zitadel.ch/oauth/v2/revoke)

> This endpoint enables clients to revoke an `acccess_token` or `refresh_token` they have been granted.

Reviewer: ```suggestion
This endpoint enables clients to revoke an `access_token` or `refresh_token` they have been granted.
```

:::important

> If you revoke an `access_token` only that specific token will be revoked. If you revoke a `refresh_token`,

Reviewer: ```suggestion
If you revoke an `access_token` only the specific token will be revoked. When revoking a `refresh_token`,
```

Author: Is it intentional that one sentence begins with "if" and the other with "when"?

Reviewer: I tried to replace the twice `If` but it sounds bad. So feel free to ignore this one 😄

Author: no that's good then, sounds better. just was not sure if it was intended

the corresponding `access_token` will be revoked as well.
:::

| Parameter | Description                      |
| --------- | -------------------------------- |
| token     | An access token or refresh token |

Depending on your authorization method you will have to provide additional parameters or headers:

<Tabs
    groupId="token-auth-methods"
    defaultValue="client_secret_basic"
    values={[
        {label: 'client_secret_basic', value: 'client_secret_basic'},
        {label: 'client_secret_post', value: 'client_secret_post'},
        {label: 'none (PKCE)', value: 'none'},
        {label: 'private_key_jwt', value: 'private_key_jwt'},
    ]}
>
<TabItem value="client_secret_basic">

> Send your `client_id` and `client_secret` as Basic Auth Header. Check [Client Secret Basic Auth Method](authn-methods#client-secret-basic) on how to build it correctly.

Reviewer: ```suggestion
Send your `client_id` and `client_secret` as Basic Auth Header. Check [Client Secret Basic Auth Method](authn-methods#client-secret-basic) on how to construct a request correctly.
```

</TabItem>
<TabItem value="client_secret_post">

Send your `client_id` and `client_secret` as parameters in the body:

| Parameter     | Description                      |
| ------------- | -------------------------------- |
| client_id     | client_id of the application     |
| client_secret | client_secret of the application |

</TabItem>
<TabItem value="none">

Send your `client_id` as parameters in the body:

| Parameter | Description                  |
| --------- | ---------------------------- |
| client_id | client_id of the application |

</TabItem>
<TabItem value="private_key_jwt">

> Send a `client_assertion` as JWT for us to validate the signature against the registered public key.

Reviewer: ```suggestion
Send a `client_assertion` as JWT for ZITADEL to verify the signature against the registered public key.
```

| Parameter             | Description                                                                                                 |
| --------------------- | ----------------------------------------------------------------------------------------------------------- |

> | client_assertion      | JWT built and signed according to [Using JWTs for Client Authentication](authn-methods#client-secret-basic) |

Reviewer: ```suggestion
| client_assertion      | JWT created and signed according to [Using JWTs for Client Authentication](authn-methods#client-secret-basic) |
```
